.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,18 @@
+name: Run tests
+on: push
+
+jobs:
+  test:
+    name: Compile and test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Cache SBT & ivy cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.ivy2/cache
+            ~/.sbt
+          key: ${{ runner.os }}-sbt-ivy-cache-${{ hashFiles('**/build.sbt') }}-${{ hashFiles('**/plugins.sbt') }}-${{ hashFiles('**/build.properties') }}
+      - name: Compile and test
+        run: sbt +test
```

Comment:
The circleCI workflow executes the command 
```
sbt -J-Xms512m -J-Xmx2g -J-XX:+UseConcMarkSweepGC -J-XX:ReservedCodeCacheSize=256m -J-XX:+CMSClassUnloadingEnabled -Djava.awt.headless=true -no-colors -batch "+test"
```
I left out all of these flags just to keep it simple, but wouldn't mind adding some/all back.

Reply:
The flags have been added on purpose, so I assume they were needed in the past.
Consider to add the options into the `.sbtopts` file to have a clean separation.

Reply:
Please also add scapegoat check.

Reply:
> Consider to add the options into the .sbtopts file to have a clean separation.

Alright, done in c9a7d0f

> Please also add scapegoat check

Scapegoat job was added. It failed with two errors because of variable shadowing, which I fixed in 1848a8e (but I actually prefer how it was previously written)

Reply:
I would suggest to disable this inspection via this sbt setting:
```
scapegoatDisabledInspections := Seq("VariableShadowing")
```

Reply:
Good idea, thanks! Disabled this inspection and reverted the previous code changes in 44d4315